log4j-api/src/main/java/org/apache/logging/log4j/spi/AbstractLogger.java

```diff
@@ -2755,77 +2755,114 @@ public void warn(final String message, final Object p0, final Object p1, final O
      * @since 3.0
      */
     @Override
-    public  LogBuilder atTrace() {
+    public  LogBuilder trace() {
         if (isTraceEnabled()) {
-            return logBuilder.get().setLevel(Level.TRACE);
+            DefaultLogBuilder builder = logBuilder.get();
+            if (builder.isInUse()) {
+                return new DefaultLogBuilder(this);
```

Comment:
Wouldn't we need to call `setLevel(Level.TRACE)` on `new DefaultLogBuilder(this)` before returning it?

Reply:
Yes, unless a second constructor is added that accepts the level.